new/rfcXXX-policyfile-endpoint.md

```diff
@@ -0,0 +1,367 @@
+relational queries, and documents are indexed for search as data bag items.
+
+This RFC describes a new Policyfile HTTP Resource API that remediates these
+issues.
+
+## Motivation
+
+    As a Chef user,
+    I want Policyfiles to have a dedicated HTTP Resource API,
+    so that Policyfiles will be validated upon upload
+    and I can query the relationships between Policyfiles and other objects,
+    and Policyfiles will not be indexed as an incorrect data type.
+
+## Specification
+
+### General Concepts
+
+A Policyfile lock is a JSON document that specifies a node's run list and the
+set of cookbooks a node will use to converge the node.
+
+A Policy Group is a set of nodes defined by a single token. Nodes must be a
+member of a Policy Group in order to use the Policyfile feature, and nodes may
+only be a member of one policy group.
+
+### Policyfile Lock
+
+A Policyfile lock is a JSON document that has the following structure:
+
+```json
+{
+  "revision_id": "edd40c30c4e0ebb3658abde4620597597d2e9c17",
+  "name": "some_policy_name",
+  "run_list": [
+    "recipe[policyfile_example_cookbook::default]"
+  ],
+  "cookbook_locks": {
+    "policyfile_example_cookbook": {
+      "version": "1.0.0",
+      "identifier": "f04cc40faf628253fe7d9566d66a1733fb1afbe9"
+    }
+  }
+}
+
+```
+
+#### `revision_id` Field (required)
+
+The `revision_id` field is used to distinguish revisions of a Policyfile lock
+document. It is a JSON String. It may contain alphanumeric characters, hyphens,
+underscores, the dot character and the colon character. In regular expression
+form: `/^[\-[:alnum:]_\.\:]+$/`. The revision ID is limited to a maximum of 255
+characters. The Chef Server must reject a Policyfile lock document when the
+revision ID does not meet these requirements.
+
+In ChefDK the revision will be calculated by hashing other content in the
+Policyfile lock document and thus will be a hexidecimal number. In order to
+allow alternate implementations to generate revision IDs according to different
+schemes (perhaps timestamps or Ci build numbers), this field will not be
+required to be a hexidecimal string.
+
+#### `name` Field (required)
+
+The `name` field describes the functional role that is fulfilled by a node that
+applies this policy. It may contain alphanumeric characters, hyphens,
+underscores, the dot character and the colon character. In regular expression
+form: `/^[\-[:alnum:]_\.\:]+$/`. The name is also limited to a maximum of 255
+characters. The Chef Server must reject a Policyfile lock document if the name
+does not meet these criteria.
+
+#### `run_list` Field (required)
+
+The `run_list` field gives the ordered list of recipes that Chef client should
+apply to converge a node. Each item MUST be in fully qualified form, that is,
+`recipe[COOKBOOK_NAME::RECIPE_NAME]`. Roles are not valid run list items in the
+Policyfile lock document, and the Chef Server MUST reject run lists that
+contain roles.
```

Comment:
It seems like this could be a point of contention for some users.  In the README you said role support was coming - was that idea dropped?  Could you give some explanation here of why roles are not supported and whether they will be in the future?